### ISSUE TITLE
lxd service disables AppArmor support because it cannot find apparmor_parser

### ISSUE THREAD

**Issue by stefano-m** [NixOS/nixpkgs#56535]
Cc @mt-caret 

## Issue description

When enabling lxd, I can see the following line in the logs:

```
 lxd[10032]: lvl=warn msg="AppArmor support has been disabled because 'apparmor_parser' couldn't be found"
```

which suggest that lxd will not be using apparmor.

However, the apparmor service starts successfully (although the second line below seems a bit odd):

```
systemd[1]: Starting apparmor.service...
apparmor_parser[9618]: Replacement succeeded for "/usr/bin/lxc-start".
apparmor_parser[9620]: Replacement succeeded for "lxc-container-default".
apparmor_parser[9620]: Replacement succeeded for "lxc-container-default-cgns".
apparmor_parser[9620]: Replacement succeeded for "lxc-container-default-with-mounting".
apparmor_parser[9620]: Replacement succeeded for "lxc-container-default-with-nesting".
apparmor_parser[9622]: Replacement succeeded for "/run/wrappers/bin/ping".
systemd[1]: Started apparmor.service.
```

### Steps to reproduce

Add

```
virtualisation.lxd.enable = true;
```

to `/etc/nixos/configuration.nix`

Then run

```shell
nixos-rebuild test
```

Then check the logs with

```shell
journalctl -u lxd -u apparmor
```

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.103, NixOS, 18.09.2265.8e70d4bd7d5 (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.1.3`
 - channels(root): `"nixos-18.09.2265.8e70d4bd7d5"`
 - channels(user): `"nixos-18.09.2265.8e70d4bd7d5, unstable-19.03pre170546.0f43790d2ce"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by joachifm** [comment]
@stefano-m you could try addding `pkgs.apparmor-utils` to the lxd service's path. As long as the profiles are loaded, however, it shouldn't matter, unless lxd uses the presence of apparmor_parser to determine whether to load additional profiles after the fact.

**Comment by stefano-m** [comment]
Thanks @joachifm , will try. If that works, would it be worth adding the package to the nixos module?

**Comment by joachifm** [comment]
@stefano-m if it fixes the issue, sure. Feel free to ping me in the PR if you get around to opening one.

**Comment by andersk** [comment]
This was fixed by #50831, it was just never backported to 18.09.

**Comment by stefano-m** [comment]
Didn't have time to verify this and not I'm not on 18.09 any longer.
Closing.
